File: src/scala/y2019/Day1.scala
package y2019

object Day1 extends Input {
  def main(args: Array[String]): Unit = {
    val input = readLines("input/2019/day1").map(_.toInt)

    val partA = input.fold(0)(_ + calcFuel(_))
    println(s"Part A: $partA") // 3297866

    val partB = input.fold(0)(_ + calcFuelComp(_))
    println(s"Part B: $partB") // 4943923
  }

  def calcFuelComp(mass: Int): Int = {
    val fuel = calcFuel(mass)
    if (fuel > 0) fuel + calcFuelComp(fuel)
    else 0
  }

  def calcFuel(mass: Int): Int = {
    val tmp = (mass / 3) - 2
    if (tmp < 0) 0
    else tmp
  }
}

File: src/python/y2019/day1.py
import os


def calc_fuel(mass):
    fuel = (mass // 3) - 2
    if fuel < 0:
        return 0
    else:
        return fuel


def calc_fuel_com(mass):
    fuel = calc_fuel(mass)
    if fuel > 0:
        return fuel + calc_fuel_com(fuel)
    else:
        return fuel


path = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../input/2019/day1"))
with open(path, "tr") as file:
    lines = list(map(int, file.read().splitlines()))

fuel_part_a = 0
fuel_part_b = 0
for mass in lines:
    fuel_part_a += calc_fuel(mass)
    fuel_part_b += calc_fuel_com(mass)

print("Part A: {}".format(fuel_part_a))  # 3297866
print("Part B: {}".format(fuel_part_b))  # 4943923
